src/unordered.rs

```diff
@@ -952,7 +963,125 @@ impl Plan {
     }
 }
 
-#[cfg(any(test, feature = "serde"))]
+impl MonomialPlan {
+    /// Returns a new FFT plan for the given vector size, selected by the provided `base_n` from a
+    /// compatible [`Plan`].
+    ///
+    /// # Panics
+    ///
+    /// - Panics if `n` is not a power of two.
+    /// - Panics if `base_n` is not a power of two.
+    /// - Panics if `base_n > n`.
```

Comment:
I think Thomas indicated try_new as a possible alternative when new can fail ?

Reply:
i don't think `try_new` is needed here. since the failure would be as a result of a logic error in the code (the user didn't pick `base_n` from an fft plan), so i don't see how we could recover from it

Reply:
fine by me then !